gmcs/linglib/yes_no_questions.py

```diff
@@ -107,49 +136,116 @@ def customize_yesno_questions(mylang, ch, rules, lrules, hierarchies):
+             [ SYNSEM.LOCAL [ CONT.HOOK.INDEX.SF ques ] ].'''
+            mylang.add(typedef, comment, section='complex')
+            supertype = 'qpart-lex-item'
+
+            # ERB 2010-04-15 If we have a finite/non-finite distinction in the
+            # language, the qpart should insist on attaching to finite clauses
+            # only.  An alternative would be to have it raise the FORM value, but
+            # I don't see any evidence for that just now. Using presence of 'form'
+            # in hierarchies to detect this situation. This could break if someone
+            # named their own feature "form", but the name-space validation should
+            # catch that.  This works because customize_form() is called before
+            # customize_yesno_questions. This is an example of cross-library
+            # interaction.
+            if 'form' in hierarchies:
+                mylang.add('qpart-lex-item := [ SYNSEM.LOCAL.CAT.VAL.COMPS.FIRST.LOCAL.CAT.HEAD.FORM finite ].')
+
+        # OZ: Second position  clitics are treated as modifiers:
+        elif ch.get('q-part-order') == 'second':
+            mylang.add('''basic-binary-phrase :+ [ SYNSEM [ L-PERIPH #periph,
+                                  NON-LOCAL.YNQ.APPEND < #ynq1,
+                                    #ynq2 > ],
+             ARGS < [ SYNSEM [ L-PERIPH #periph,
+                               NON-LOCAL.YNQ #ynq1 ] ],
+                    [ SYNSEM [ L-PERIPH -,
+                               NON-LOCAL.YNQ #ynq2 ] ] > ].''',section='addenda')
+            mylang.add('basic-head-mod-phrase-simple :+ '
+                       '[ HEAD-DTR.SYNSEM.L-PERIPH #periph, '
+                       'NON-HEAD-DTR.SYNSEM.LOCAL.CAT.HEAD.MOD < [ L-PERIPH #periph ] > ].',section='addenda')
+            mylang.add(SAME_PERIPH,section='phrases')
+            mylang.add(SAME_YNQ,section='phrases')
+            mylang.add('bare-np-phrase := same-periph-unary-phrase & same-ynq-unary-phrase & [ SYNSEM.LIGHT - ].')
+            comment = 'Second position question particles are treated as modifiers.'
+            typedef = lexbase.QUES_CLITIC
+            mylang.add(typedef, comment, section='complex')
+            supertype = 'ques-clitic-lex'
+            # Constrain all other words to be YNQ-empty:
+            mylang.add('non-ynq-word := word-or-lexrule & [ SYNSEM.NON-LOCAL.YNQ.LIST < > ].')
+            mylang.add('basic-zero-arg :+ non-ynq-word.',section='addenda')
+            mylang.add('basic-one-arg :+ non-ynq-word.',section='addenda')
+            mylang.add('basic-two-arg :+ non-ynq-word.',section='addenda')
+            mylang.add('basic-three-arg :+ non-ynq-word.',section='addenda')
+            mylang.add('intersective-mod-lex :+ non-ynq-word.',section='addenda')
+            roots.add('root := [ SYNSEM.NON-LOCAL.YNQ.LIST < > ].')
+            mylang.add(INT_CL,section='phrases')
+            mylang.add(DECL_CL,section='phrases')
+            rules.add('int-cl := int-cl.')
```

Comment:
What does?

Reply:
`int-cl` is its own supertype? How do these grammars even compile with ACE? Maybe the line is not used in our tests?

Reply:
That's rules though; I don't think there is a restriction of this sort there? That's just a name for the rule to be used? I mean I can rename it but I don't think it is causing any problems.

Reply:
Ah ok, so it's not creating a type. I was unaware that rules and types have separate namespaces (or do they?). Perhaps we should adopt some more thorough naming conventions for rules and types like the ERG has, because even if this works it is still a it confusing.

Reply:
I can rename it if that's confusing. Didn't occur to me.

Reply:
So `int-cl` isn't its own supertype? The LHS is a definition in `rules.tdl` and the RHS is a type specified in `mylang.tdl`? I agree this is confusing, and renaming would be helpful if possible.